ClientExecutorServiceTest.testGetFutureAfterCancel

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-pr-builder/com.hazelcast$hazelcast-client/17278/console

```
Tests run: 15, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 23.423 sec <<< FAILURE! - in com.hazelcast.client.executor.ClientExecutorServiceTest
testGetFutureAfterCancel(com.hazelcast.client.executor.ClientExecutorServiceTest)  Time elapsed: 21.842 sec  <<< FAILURE!
java.lang.AssertionError: Expected exception: java.util.concurrent.CancellationException
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:32)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:132)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastParallelClassRunner.access$301(HazelcastParallelClassRunner.java:46)
	at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:139)
	at java.lang.Thread.run(Thread.java:662)
```

--- a/hazelcast/src/main/java/com/hazelcast/executor/impl/DistributedExecutorService.java
+++ b/hazelcast/src/main/java/com/hazelcast/executor/impl/DistributedExecutorService.java
@@ -211,8 +211,8 @@ public class DistributedExecutorService implements ManagedService, RemoteService
                 if (uuid != null) {
                     submittedTasks.remove(uuid);
                 }
-                sendResponse(result);
                 if (!isCancelled()) {
+                    sendResponse(result);
                     finishExecution(name, Clock.currentTimeMillis() - start);
                 }
             }

